airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/record_buffer/BaseSerializedBuffer.kt

```diff
@@ -42,7 +42,7 @@ protected constructor(private val bufferStorage: BufferStorage) : SerializableBu
      */
     @Deprecated("")
     @Throws(IOException::class)
-    protected abstract fun writeRecord(record: AirbyteRecordMessage)
+    protected abstract fun writeRecord(record: AirbyteRecordMessage, generationId: Long = 0)
 
     /**
      * TODO: (ryankfu) move destination to use serialized record string instead of passing entire
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 62]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Other consumers using the Async rely on this method. We were unable to delete the method on L45 because of S3 not moving to Async and remnants of old calls. Any chance we can achieve that in this PR ?<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42409?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Same comment as above.